ehitchhiking-ui/src/app/shared/modals/profile-modal/profile-modal.component.ts

```diff
@@ -27,11 +22,17 @@ export class ProfileModalComponent implements OnInit {
 		public dialogRef: MatDialogRef<ProfileModalComponent>,
 		private formBuilder: FormBuilder,
 		private carsInfoService: CarsInfoService,
-		private apiService: ApiService
+		private profileModalService: ProfileModalService
 	) {}
 
 	ngOnInit(): void {
-		this.carsInfoForm = this.carsInfoService.toFormGroup(this.user.cars, this.formBuilder);
+		this.isLoading = true;
+		this.currentUser = this.profileModalService.getCurrentUser();
+		this.profileModalService.getCarsList(this.currentUser.id).then((cars) => {
+			this.currentUser.cars = cars;
+			this.carsInfoForm = this.carsInfoService.toFormGroup(cars, this.formBuilder);
+			this.isLoading = false;
+		});
```

Comment:
pls use finally block to disable loading

then((cars) => {...}).finally(() => {this.isLoading = false;})